pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go

```diff
@@ -164,15 +179,40 @@ func (c *SyncController) cleanUpWorks(ctx context.Context, namespace, name strin
 	return errors.NewAggregate(errs)
 }
 
-func (c *SyncController) buildWorks(ctx context.Context, quota *policyv1alpha1.FederatedResourceQuota, clusters []clusterv1alpha1.Cluster) error {
+func (c *SyncController) cleanUpOrphanWorks(ctx context.Context, quota *policyv1alpha1.FederatedResourceQuota) error {
+	var errs []error
+	workList := &workv1alpha1.WorkList{}
+	if err := c.List(ctx, workList, client.MatchingLabels{
+		util.FederatedResourceQuotaNamespaceLabel: quota.GetNamespace(),
+		util.FederatedResourceQuotaNameLabel:      quota.GetName(),
+	}); err != nil {
+		klog.Errorf("Failed to list works, err: %v", err)
+		return err
+	}
+
+	for index := range workList.Items {
+		work := &workList.Items[index]
+		if !isOrphanQuotaWork(work, quota.Spec) {
+			continue
+		}
+		if err := c.Delete(ctx, work); err != nil && !apierrors.IsNotFound(err) {
+			klog.Errorf("Failed to delete work(%s): %v", klog.KObj(work).String(), err)
+			errs = append(errs, err)
+		}
+	}
+
+	return errors.NewAggregate(errs)
+}
+
+func (c *SyncController) buildWorks(ctx context.Context, quota *policyv1alpha1.FederatedResourceQuota, clusters []string) error {
 	var errs []error
 	for _, cluster := range clusters {
 		resourceQuota := &corev1.ResourceQuota{}
 		resourceQuota.APIVersion = "v1"
 		resourceQuota.Kind = "ResourceQuota"
 		resourceQuota.Namespace = quota.Namespace
 		resourceQuota.Name = quota.Name
-		resourceQuota.Spec.Hard = extractClusterHardResourceList(quota.Spec, cluster.Name)
```

Comment:
It's quite easy to ignore the non-existent cluster in staticAssignments by checking the return value of `extractClusterHardResourceList(quota.Spec, cluster.Name)`.

I mean we don't need `extractExistedClusters()`.